apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/OpenApiDocs.java

```diff
@@ -0,0 +1,183 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Set;
+import org.hsqldb.jdbc.JDBCDataSource;
+
+/** Program to gather the OpenAPI yaml content for V1 and V2 and store. */
+public class OpenApiDocs {
+
+  /** The HSQL database url. */
+  private static final String INMEM_HSQL_DATABASE_URL = "jdbc:bfd-test:hsqldb:mem";
+
+  /** The version from the project pom. */
+  private final String projectVersion;
+
+  /** The destination directory for the output of the OpenAPI yaml file documents. */
+  private final String destinationDir;
+
+  /** The base server url. */
+  private static String baseServerUrl;
+
+  /** The request specification to use with the api-docs requests. */
+  private static RequestSpecification requestAuth;
+
+  /**
+   * Constructs an instance of OpenApiDocs for a given project version and destination directory.
+   *
+   * @param projectVersion @see #projectVersion
+   * @param destinationDir @see #destinationDir
+   */
+  private OpenApiDocs(String projectVersion, String destinationDir) {
+    this.projectVersion = projectVersion;
+    this.destinationDir = destinationDir;
+  }
+
+  /**
+   * Downloads the OpenAPI yaml from the api-docs V1 endpoint using a local E2E test server
+   * instance.
+   *
+   * @param apiVersion the bfd api version, either V1 or V2.
+   * @throws IOException when IO operations fail
+   */
+  void downloadApiDocs(String apiVersion) throws IOException {
+    var apiDocsUrl = String.format("%s/%s/fhir/api-docs", baseServerUrl, apiVersion.toLowerCase());
+    var response =
+        given()
+            .spec(requestAuth)
+            .when()
+            .get(apiDocsUrl)
+            .then()
+            .contentType("text/yaml")
+            .extract()
+            .asString();
+
+    writeFile(apiVersion, response);
+  }
+
+  /**
+   * Write the OpenAPI yaml content to a file.
+   *
+   * @param apiVersion either V1 or V2
+   * @param contents the OpenAPI yaml contents to write to the file
+   * @throws IOException when file operations fail
+   */
+  private void writeFile(String apiVersion, String contents) throws IOException {
+    var fileName =
+        String.format("%s/%s-OpenAPI-%s.yaml", destinationDir, apiVersion, projectVersion);
+    try (var fileWriter = new FileWriter(fileName)) {
+      fileWriter.write(contents);
+    }
+  }
+
+  /**
+   * Entry point, starts an E2E test server instance and downloads OpenAPI yaml.
+   *
+   * @param args The project version, working directory, and destination directory.
+   */
+  public static void main(String[] args) {
+
+    // Validate arguments.
+    validateArgs(args);
+
+    // Update working directory so E2E test server instance can find properties.
+    System.setProperty("user.dir", args[1]);
+
+    // Set database url for in memory HSQL database
+    System.setProperty("its.db.url", INMEM_HSQL_DATABASE_URL);
+
+    var openApiDocs = new OpenApiDocs(args[0], args[2]);
```

Comment:
It would be best to extract the arguments into variables so they have meaningful names early on.  Spreading use of the array and its indicies around makes the code harder to maintain.